airflow/models/xcom_arg.py

```diff
@@ -388,6 +388,9 @@ class _ZipResult(Sequence):
     def __init__(self, values: Sequence[Sequence | dict], *, fillvalue: Any = NOTSET) -> None:
         self.values = values
         self.fillvalue = fillvalue
+        # use the generator here, rather than in __len__ to improve efficiency
+        lengths = (len(v) for v in self.values)
+        self.length = min(lengths) if isinstance(self.fillvalue, ArgNotSet) else max(lengths)
```

Comment:
I don’t think it is very meaningful to do this. IIRC I wrote the `__len__` implementation as it previously is because the length is checked very infrequently and usually zero to one times in each process (usually in a task worker), so caching like this actually makes it consume more resource than putting the logic in `__len__`. Would you mind reverting this part? It is not really related to the overall fix and shouldn’t block it.

Reply:
Just saw this.  Sure.  I can reverse it.